mathics/builtin/strings.py

```diff
@@ -699,3 +699,109 @@ class StringQ(Test):
 
     def test(self, expr):
         return isinstance(expr, String)
+        
+      
+class StringTake(Builtin):
+    """
+    StringTake["string",n] gives the first n characters in "string"
+    StringTake["string",-n] gives the last n characters in "string"
+    StringTake["string",{m,n}] gives characters m through n in "string"
+    >> StringTake["abc", 2]
+     = ab
+    >> StringTake["abc", -2]
+     = bc
+    >> StringTake["abcd", {2,3}]
+     = bc
+    """
+    messages = {
+        'strse': 'String expected at position `1` in `2`.',
+        'naioli': 'Integer or list of two Intergers are expected at position `2` in `2`.',
+        'badrange': 'It is not possible to take the substring for elements `2` in `2`.',
+    }
+    
+    def apply1_(self, string, n, evaluation):
+        'StringTake[string_,n_]'
+        if not isinstance(string, String):
+            return evaluation.message('StringTake', 'strse', string)
+        if isinstance(n, Integer):
+            pos=n.value
+            if pos>0:
+                return String(string.get_string_value()[:pos])
+            if pos<0:
+                return String(string.get_string_value()[pos:])
+        return evaluation.message('StringTake', 'naioli', n)
+
+
+    def apply2_(self, string, ni,nf, evaluation):
+        'StringTake[string_,{ni_,nf_}]'
+        if not isinstance(string, String):
+            return evaluation.message('StringTake', 'strse', string)
+        if isinstance(ni, Integer) and  isinstance(nf, Integer)  :
+            if ni.value==0 or nf.value==0:
+                return evaluation.message('StringTake', 'badrange', ni,nf)
+            fullstring=string.get_string_value()
+            posi=ni.value        
+            if posi<0:
+                posi=len(fullstring)+posi+1
+            posf=nf.value
```

Comment:
Is the aim to have `0 < posi < posf <= len(fullstring)` ? There might be some more edge cases to consider here. e.g. `ni=-10` and `fullstring="ab"`. The solution to this case should probably be to raise an error